### PR TITLE
:sparkles: add python client

### PR DESCRIPTION
**Pull request by roblen45**
**Changes:**

1. :sparkles: python client added

PR: #8 

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: python/Pipfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
```
Comment (cmg-dev) [inline python/Pipfile]: Richtige Angaben müssen hier noch hin.

File: python/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# ExternalTaskApiClient written in python
+
+This is an ExternalTaskApiClient written in python.
```
Comment (cmg-dev) [inline python/README.md]:
```suggestion
This is an ExternalTaskApiClient written in Python.
```

File: python/external_task_api_client/external_task_api_client_service.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import asyncio
+import json
+
+
+class ExternalTaskApiClientService:
+
+    def __init__(self, baseUrl):
+        self.__baseUrl = baseUrl
+
+    async def extendLock(self, identity, workerId, externalTaskId, additionalDuration):
```
Comment (cmg-dev) [inline python/external_task_api_client/external_task_api_client_service.py]: Snake_Case in Python nutzen.

```suggestion
    async def extend_lock(self, identity, worker_id, external_task_id, additional_duration):
```

File: python/external_task_api_client/external_task_api_client_service.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        uri = "task/" + externalTaskId + "/extend_lock"
```
Comment (cmg-dev) [inline python/external_task_api_client/external_task_api_client_service.py]:
```suggestion
        uri = 'task/{}/extend_lock'.format(external_task_id)
```

So würde man das "pythonic" machen.

File: python/external_task_api_client/external_task_api_client_service.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        request = {
+            "workerId": workerId,
+            "additionalDuration": additionalDuration
```
Comment (cmg-dev) [inline python/external_task_api_client/external_task_api_client_service.py]:
```suggestion
            "additionalDuration": additional_duration
```

File: python/external_task_api_client/external_task_api_client_service.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+class ExternalTaskApiClientService:
+
+    def __init__(self, baseUrl):
+        self.__baseUrl = baseUrl
+
+    async def extendLock(self, identity, workerId, externalTaskId, additionalDuration):
+        uri = "task/" + externalTaskId + "/extend_lock"
+        request = {
+            "workerId": workerId,
```
Comment (cmg-dev) [inline python/external_task_api_client/external_task_api_client_service.py]:
```suggestion
            "workerId": worker_id,
```

File: python/external_task_api_client/external_task_api_client_service.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+            "additionalDuration": additionalDuration
+        }
+
+        return await self.__sendPostToExternalTaskApi(identity, uri, request)
```
Comment (cmg-dev) [inline python/external_task_api_client/external_task_api_client_service.py]:
```suggestion
        return await self.__send_post_to_external_task_api(identity, uri, request)
```

File: python/external_task_api_client/external_task_api_client_service.py [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+    async def fetchAndLockExternalTasks(self, identity, workerId, topicName, maxTasks, longPollingTimeout, lockDuration):
```
Comment (cmg-dev) [inline python/external_task_api_client/external_task_api_client_service.py]:
```suggestion
    async def fetch_and_lock_external_tasks(self, identity, worker_id, topic, max_tasks, long_polling_timeout, lock_duration):
```

Ich würde `topicName` zu `topic` ändern, da es anderswo nur als topic auftaucht.

File: python/external_task_api_client/external_task_api_client_service.py [hunk 9]
```diff
@@ -0,0 +1,71 @@
+import aiohttp
+import asyncio
+import json
+
+
+class ExternalTaskApiClientService:
+
+    def __init__(self, baseUrl):
+        self.__baseUrl = baseUrl
```
Comment (cmg-dev) [inline python/external_task_api_client/external_task_api_client_service.py]: Die URIs kann man gut hier ablegen, auch mit format string:

```suggestion
        self.__baseUrl = baseUrl
        self.__extend_lock_base_uri =  'task/{}/extend_lock'
```

Und dann kann man das so benutzen:

```python
uri = self.__extend_lock_base_uri.format(external_task_id)
```

Comment (cmg-dev) [reply python/external_task_api_client/external_task_api_client_service.py]: wenn man alle routen hier sammelt, macht es das einfach, die zu ändern/überblicken.

File: python/external_task_api_client/external_task_api_client_service.py [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+        uri = "fetch_and_lock"
+        request = {
+            "workerId": workerId,
+            "topicName": topicName,
+            "maxTasks": maxTasks,
+            "longPollingTimeout": longPollingTimeout,
+            "lockDuration": lockDuration
+        }
+
+        return await self.__sendPostToExternalTaskApi(identity, uri, request)
```
Comment (cmg-dev) [inline python/external_task_api_client/external_task_api_client_service.py]:
```suggestion
        return await self.__send_post_to_external_task_api(identity, uri, request)
```

File: python/Pipfile (hunk 1 above)
Comment (roblen45) [reply python/Pipfile]: Was muss denn dort rein? Das sind doch die nötigen Angaben oder nicht?
Könnte zumindest damit publishen

**Pull request merged**